concordium-std/src/impls.rs

```diff
@@ -1893,6 +1918,79 @@ fn parse_call_response_code(code: u64) -> CallContractResult<ExternCallResponse>
     }
 }
 
+/// Check wether the response code from calling `invoke` is encoding a failure,
+/// i.e. whether the first 3 bytes are all set.
+#[inline(always)]
+fn is_invoke_failure_code(code: u64) -> bool {
+    code & 0xffff_ff00_0000_0000 == 0xffff_ff00_0000_0000
+}
+
+/// Decode the account balance response code.
+///
+/// This is necessary since Wasm only allows us to pass simple scalars as
+/// parameters. Everything else requires passing data in memory, or via host
+/// functions, both of which are difficult.
+///
+/// The response is encoded as follows.
+/// - success is encoded as 0
+/// - missing account is encoded as '0xffff_ff02_0000_0000'
+fn parse_query_account_balance_response_code(
+    code: u64,
+) -> Result<ExternCallResponse, QueryAccountBalanceError> {
+    if is_invoke_failure_code(code) {
+        if let 0xffff_ff02_0000_0000 = code {
+            return Err(QueryAccountBalanceError);
+        } else {
+            unsafe { crate::hint::unreachable_unchecked() }
+        };
+    }
```

Comment:
The main reason is that, in the node, we reuse the return value logic from the other invoke calls, which defines a failed invocation as the first 3 bytes are set.
I am not sure what is best

Reply:
It is better to hard fail if there is a bug either in the node or here.

Reply:
Okay I'll keep the check then.

I realized that the first 3 bytes are not set in case of failure for the response code passed to the contract. This is just the case for the response code passed in FFI in the node.
I have update the doc comments stating this and fixed the parsing. (Only the queries actually followed the doc comments, so the rest should behave the same)